core/domain/question_domain.py

```diff
@@ -109,11 +124,14 @@ def __init__(
             question_data_schema_version: int. The schema version for the data.
             language_code: str. The ISO 639-1 code for the language this
                 question is written in.
+            status: str. The status of the question among approved, rejected,
+                pending or private.
         """
-        self.question_id = question_id
+        self.id = question_id
         self.question_data = question_data
```

Comment:
This seems to have been an error from before, but I am concerned about question_data being a dict. Instead it should be a domain object, and we should distinguish between question_data (the object) and question_data_dict (the dict).

(If it's possible to give "question_data" a more descriptive name, that would be great too.)

EDIT: looks like it's a State object. Can we convert it to a State here?

Reply:
Changed.